interface.go

```diff
@@ -68,10 +69,15 @@ func (c *Cache) GetOrCompute(key string, evaluator func() (interface{}, error))
 		mutex.Unlock()
 		return value, nil
 	}
-	if err == ErrNotFound {
-		value, err := evaluator()
-		if err == nil {
+
+	value, evaluatorErr := evaluator()
+
+	if evaluatorErr == nil {
+		// value evaluted correctly
+		if err == ErrNotFound {
```

Comment:
If c.Get() returns a real error (not ErrNotFound), but evaluator performs fine, we return correct values + Get() error. Not sure if this is intended.

Reply:
Yes, it is. I've made a complementary PR in g2 https://github.com/datasapiens/g2/pull/887. I want to return an error and it is logged in g2. Generally such construction is used in g2:
```
result, err := dataCache.GetOrCompute(key, dataEvaluator)
if err != nil {
		if result == nil {
			return nil, err
		}
		cs.logger.Errorf("error getting valur from cache %v", err)
	}
```

Reply:
I have thought that cashier should return `value` and `error` and g2 backed should decide what to do with error